### DISCUSSION TITLE
Rules - Action value for destination / origin account - Bug or am I missing something ?

### DISCUSSION THREAD

**Discussion by top-cg** [firefly-iii/firefly-iii#5588] [Firefly III] [answered]
Hello,

I pretty much figure out how the app deals will Accounts ... and unless mistaken, she is considering "Revenues accounts" and "Assets accounts" to be "Source accounts" ... and "expenses / liab accounts" to be "Destination accounts".

Yet this rule don't seems to apply on the "Rules" ... indeed trying to imagine a good way to manage refunds, I ended in trying to duplicate my "expense accounts" into the "revenue accounts" ... i ended in a bad spot.

I have a rule in front of me and when i'm trying to add a value into "set destination account" ... the list based on the existing accounts in the app contains both the revenues and expense accounts ... and as they have the same name ... oups ... that's a problem.  

Am I wrong when thinking the app saw:
- Revenues accounts = Source accounts
- Assets accounts = Source accounts
- Liab accounts = Destination accounts
- Expenses accounts = Destination accounts

Is it because the "action value" list of proposals based on what you entered doesn't depends on the Action nature selected ?

[https://ibb.co/Qfx8JSW](url)

<a href="https://ibb.co/Qfx8JSW"><img src="https://i.ibb.co/xDrm29n/2022-01-22-09-30-19.png" alt="2022-01-22-09-30-19" border="0"></a><br /><a target='_blank' href='https://usefulwebtool.com/fr/convertir-texte-en-binaire'>lexilogos anglais français</a><br />

thanks for your help !

**Comment by JC5** [comment, marked as answer]
See also https://github.com/firefly-iii/firefly-iii/issues/5530

**Reply by top-cg** [reply]
Hi,

Gosh you are fast to answer ...

I read you link.

On my side I don't the issue as a real problem or a bug, or something to change in the app.
Yet I'm thinking about a possible improvement to make such behavior of the app easier to deal with ... 

Will it be possible or not uber complex to add behind the account his "nature" ?
As you may see in my print screen i made it manually by adding " (rev) " behind 1 of the 2 accounts to understand what i was seeing.

In the meantime I'm starting to think this maybe a very bad idea to have Revenues and Expenses accounts with the same name ... and therefore i should rethink it and dig more the classification side of the app.

In any case ... i'm into your app since few days only, trying to set it right before starting adding tons of data into it and there is no bottom to what you can do apparently ! I even found a way to play with tags to have a kind of OR / AND trigger. Amazing work !

**Reply by JC5** [reply]
Thanks!

The rules work out themselves which account is which, so it won't matter too much.

Glad to hear you like the tool!

**Reply by top-cg** [reply]
> The rules work out themselves which account is which, so it won't matter too much.

You got me ... issue solved.
